### PR TITLE
prometheus-smartctl-exporter: fix path to smartctl

### PR DESCRIPTION
**Pull request by peterhoeg**
## Description of changes

Been running with this patch for ages - I doubt the exporter works without it.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/monitoring/prometheus/smartctl-exporter/default.nix [hunk 1]
```diff
@@ -17,6 +18,13 @@ buildGoModule rec {
 
   vendorHash = "sha256-WUB2FgBl4Tybz7T0yvcSYIlG75NEhXpn1F0yuB9F21g=";
 
+  postPatch = ''
+    for f in main.go README.md; do
+      substituteInPlace $f \
+        --replace-fail /usr/sbin/smartctl ${lib.getBin smartmontools}/bin/smartctl
```
Comment (nevivurn) [inline pkgs/servers/monitoring/prometheus/smartctl-exporter/default.nix:24]: `smartmontools.meta.mainProgram` is already smartctl, we could just use `lib.getExe smartmontools` here.

**Review by nevivurn — COMMENTED** [review]
This also makes the overriding code in the NixOS module redundant, we should consider removing it.

https://github.com/NixOS/nixpkgs/blob/a04917482c5538ab16ce6fccbedcba555ba6c2b3/nixos/modules/services/monitoring/prometheus/exporters/smartctl.nix#L8

**Comment by peterhoeg** [comment]
Thanks for the feedback @nevivurn  - all good suggestions, all addressed.

**Comment by peterhoeg** [comment]
@GrahamcOfBorg eval

**Review by nevivurn — APPROVED** [review]
NixOS module also tested working, sgtm

File: pkgs/servers/monitoring/prometheus/smartctl-exporter/default.nix [hunk 2]
```diff
@@ -17,6 +18,13 @@ buildGoModule rec {
 
   vendorHash = "sha256-WUB2FgBl4Tybz7T0yvcSYIlG75NEhXpn1F0yuB9F21g=";
 
+  postPatch = ''
+    for f in main.go README.md; do
+      substituteInPlace $f \
+        --replace-fail /usr/sbin/smartctl ${lib.getExe smartmontools}
+    done
```
Comment (mweinelt) [inline pkgs/servers/monitoring/prometheus/smartctl-exporter/default.nix:25]:
```suggestion
    substituteInPlace main.go README.md \
      --replace-fail /usr/sbin/smartctl ${lib.getExe smartmontools}
```

Comment (peterhoeg) [reply pkgs/servers/monitoring/prometheus/smartctl-exporter/default.nix:25]: Nice, I didn't realize that `substituteInPlace` took multiple files.

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build prometheus-smartctl-exporter

**Pull request merged**